ports/servoshell/running_app_state.rs

```diff
@@ -83,6 +195,53 @@ pub trait RunningAppStateTrait {
         self.base().webdriver_receiver.as_ref()
     }
 
+    fn webview_collection(&self) -> std::cell::Ref<'_, WebViewCollection> {
+        self.base().webview_collection.borrow()
+    }
+
+    fn webview_collection_mut(&self) -> std::cell::RefMut<'_, WebViewCollection> {
+        self.base().webview_collection.borrow_mut()
+    }
+
+    /// Returns all webviews in creation order.
+    fn webviews(&self) -> Vec<(WebViewId, WebView)> {
+        self.webview_collection()
+            .all_in_creation_order()
+            .map(|(id, webview)| (id, webview.clone()))
+            .collect()
+    }
+
+    fn add_webview(&self, webview: WebView) {
+        let webview_id = webview.id();
+        self.webview_collection_mut().add(webview);
+        let total = self.webview_collection().len();
+        info!("Added webview with ID: {webview_id:?}, total webviews: {total}");
+    }
+
+    fn focused_webview(&self) -> Option<WebView> {
+        self.webview_collection().focused().cloned()
+    }
+
+    #[allow(dead_code)]
+    fn newest_webview(&self) -> Option<WebView> {
+        self.webview_collection().newest().cloned()
+    }
+
+    /// Gets the "active" webview: the focused webview if there is one,
+    /// otherwise the most recently created webview.
```

Comment:
```suggestion
    /// Gets the "active" [`WebView`]: the focused [`WebView`] if there is one,
    /// otherwise the most recently created [`WebView`].
```